### ISSUE TITLE
"saved model" or .ONNX format of PreTrained 20180402-114759 model?

### ISSUE THREAD

**Issue by FrancPS** [davidsandberg/facenet#1165]
Does someone have it, or is able to create it?

**Comment by luisfmnunes** [comment]
To create a saved model you can use the following code:

```
import tensorflow._api.v2.compat.v1 as tf 
tf.disable_v2_behavior()
from tensorflow.python.saved_model import signature_constants
from tensorflow.python.saved_model import tag_constants

# This is the core function to conversion
def covert_pb_to_server_model(pb_model_path, export_dir, input_name='input:0', output_name='output:0'):
    graph_def = read_pb_model(pb_model_path)
    covert_pb_saved_model(graph_def, export_dir, input_name, output_name)


def read_pb_model(pb_model_path):
    with tf.gfile.GFile(pb_model_path, "rb") as f:
        graph_def = tf.GraphDef()
        graph_def.ParseFromString(f.read())
        return graph_def


def covert_pb_saved_model(graph_def, export_dir, input_name='input:0', output_name='output:0'):
	builder = tf.saved_model.builder.SavedModelBuilder(export_dir)

	sigs = {}
	with tf.Session(graph=tf.Graph()) as sess:
		tf.import_graph_def(graph_def, name="")
		g = tf.get_default_graph()
		print(input_name)
		print(output_name)
		all_ops = [op.name for op in g.get_operations()]
		for name in all_ops:
			print(name)
		inp = g.get_tensor_by_name(input_name)
		out = g.get_tensor_by_name(output_name)

		sigs[signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY] = \
		tf.saved_model.signature_def_utils.predict_signature_def(
		    {"input": inp}, {"output": out})

		builder.add_meta_graph_and_variables(sess,ns()
		                                 [tag_constants.SERVING],
		                                 signature_def_map=sigs)
		builder.save()
```


just remember that the output layer name is embeddings and not outputs, and it is necessary to inform the input and output layers in the code above. This is a module, you can import it on a Jupyter notebook or your main and call the core function

**Comment by yunuscukran** [comment]
I'm sorry, but what are the actual names of the inputs and outputs? It seems to me that multiple people are stuck with this ONNX conversıon due to not knowing these parameter names.

Thanks!

**Comment by yunuscukran** [comment]
So I managed to finish the conversion, by deleting the ns() after sess --this might be a typo--.

After that, I got the following error. Any insights?

`Unable to lift tensor <tf.Tensor 'embeddings:0' shape=(None, 512) dtype=float32> because it depends transitively on placeholder <tf.Operation 'phase_train' type=Placeholder> via at least one path, e.g.: embeddings (Mul) <- embeddings/Rsqrt (Rsqrt) <- embeddings/Maximum (Maximum) <- embeddings/Sum (Sum) <- embeddings/Square (Square) <- InceptionResnetV1/Bottleneck/BatchNorm/Reshape_1 (Reshape) <- InceptionResnetV1/Bottleneck/BatchNorm/Shape (Shape) <- InceptionResnetV1/Bottleneck/MatMul (MatMul) <- InceptionResnetV1/Logits/Dropout/cond/Merge (Merge) <- InceptionResnetV1/Logits/Dropout/cond/dropout/mul (Mul) <- InceptionResnetV1/Logits/Dropout/cond/dropout/Floor (Floor) <- InceptionResnetV1/Logits/Dropout/cond/dropout/add (Add) <- InceptionResnetV1/Logits/Dropout/cond/dropout/random_uniform (Add) <- InceptionResnetV1/Logits/Dropout/cond/dropout/random_uniform/min (Const) <- InceptionResnetV1/Logits/Dropout/cond/switch_t (Identity) <- InceptionResnetV1/Logits/Dropout/cond/Switch (Switch) <- phase_train (Placeholder)`
